lib/classes/board.rb

```diff
@@ -0,0 +1,77 @@
+module OurTicTacToe
+	class Board
+		attr_reader :grid
+		def initialize(grid_board = {})
+			@grid = grid_board.fetch(:grid, default_board)
+		end
+
+		def get_cell(row, col)
+			grid[row][col]
+		end
+
+		def set_cell(row, col, value)
+			cell = get_cell(row, col)
+			cell.value.empty? ? cell.value = value : nil
```

Comment:
After considering this suggestion we decided not to go with it for the following reasons:
1. We don't want to overwrite the cell if it has a value
2. We need this method to return nil if its not empty because our #play method requires it.
```
move = board.set_cell(row, col, @current_player.turn)
    		while move.nil?
    			puts solicit_move + ' Please enter a number in an empty cell.'
					col, row = get_move
					puts ''
    			move = board.set_cell(row, col, @current_player.turn)
    		end
```

Reply:
I'll explain myself better.
1) First of all, for cell.value I don't think you need to store the empty string. Why not initialize cell value to `nil`? 
2) If the previous change was implemented you could do something like
```
    def set_cell(row, col, value)
      cell = get_cell(row, col)
      # Write cell.value only if it's nil
      cell.value ||= value
      # Return if the cell value is the same as the one you got as argument. This method will return true if the wirte was succesful and false otherwise
      cell.value == value
      # At this point i'm not sure why do you need this method to return nil
    end
```

Reply:
> We don't want to overwrite the cell if it has a value

That's exactly what the `||=` operator does

Reply:
Thanks, I actually really like what we did.

tried ||= does not seem to set the value, I don't know why. hmmm....

Reply:
Oh right, because the initial value of the cell is ""

Reply:
I will initialize the cell value to nil as you suggested. It would make some breaking changes but I guess we can do that to accommodate your suggestion! thanks :)

Reply:
Wow, this change is causing problems, for some reason, the function does not return anything other than what is returned at ||= statement event when we explicitly say return cell.value == value it still returns the value or nil. Weird. I think we are missing something but perhaps we can make an issue out of this and move on since it does not affect functionality and is a matter of personal preference.

We will leave it as it was before.

See iss #17

Reply:
See commit [4d1f2abce5779b64ea7bf3ad9d38d989c06aafd9](https://github.com/adriaanbd/tic-tac-toe/pull/18/commits/4d1f2abce5779b64ea7bf3ad9d38d989c06aafd9) to see that we attempted.

We reverted back to the original version in commit [a90931c9cbcf66e3b33aa777b90d3557fee8fbf1](https://github.com/adriaanbd/tic-tac-toe/pull/18/commits/a90931c9cbcf66e3b33aa777b90d3557fee8fbf1)